packages/flutter/lib/src/services/binding.dart

```diff
@@ -40,9 +40,32 @@ abstract class ServicesBinding extends BindingBase {
   }
 
   Stream<LicenseEntry> _addLicenses() async* {
-    final String rawLicenses = await rootBundle.loadString('LICENSE', cache: false);
-    final List<LicenseEntry> licenses = await compute(_parseLicenses, rawLicenses, debugLabel: 'parseLicenses');
-    yield* new Stream<LicenseEntry>.fromIterable(licenses);
+    // We use timers here (rather than scheduleTask from the scheduler binding)
+    // because the services layer can't use the scheduler binding (the scheduler
+    // binding uses the services layer to manage its lifecycle events). Timers
+    // are what scheduleTask uses under the hood anyway. The only difference is
+    // that these will just run next, instead of being prioritized relative to
+    // the other tasks that might be running. Using _something_ here to break
+    // this into two parts is important because isolates take a while to copy
+    // data at the moment, and if we receive the data in the same event loop
+    // iteration as we send the data to the next isolate, we are definitely
+    // going to miss frames. Another solution would be to have the work all
+    // happen in one isolate, and we may go there eventually, but first we are
+    // going to see if isolate communication can be made cheaper.
+    // See: https://github.com/dart-lang/sdk/issues/31959
+    //      https://github.com/dart-lang/sdk/issues/31960
+    // TODO(ianh): Remove this complexity once these bugs are fixed.
+    final Completer<String> rawLicenses = new Completer<String>();
+    Timer.run(() async {
+      rawLicenses.complete(rootBundle.loadString('LICENSE', cache: false));
+    });
+    await rawLicenses.future;
```

Comment:
I think this is the same as `new Future(... logic ...)`. Does not require a completer.

Reply:
Not sure I understand what you mean. Can you show a snippet of how you'd use `new Future` to do this?

Reply:
I may be missing something, but was thinking this:

```dart
// The callback passed to `new Future` is passed to Timer.run.
final String rawLicenses = await new Future<String>(() => rootBundle.loadString('LICENSE', cache: false));
final List<LicenseEntry> parsedLicenses = await new Future<List<LicenseEntry>>(
  () => compute(_parseLicenses, await rawLicenses.future, debugLabel: 'parseLicenses'),
);
```

Reply:
[update: fixed the code a little]

Reply:
If you don't use `Timer.run`, you're using microtasks, which all run before the next event loop iteration, and thus it becomes impossible for the engine to render a frame between the first future `compute` completing the future and the second starting its isolate, both of which are very expensive operations (see the bugs linked to in the comment).

Reply:
Or to put it another way, there's no difference between your code and:
```dart
final String rawLicenses = await rootBundle.loadString('LICENSE', cache: false);
final List<LicenseEntry> parsedLicenses = await compute(_parseLicenses, rawLicenses, debugLabel: 'parseLicenses');
```
...which is what it was before. :-)

Reply:
Yep, `new Future(...)` uses `Time.run` (at least, according to the [docs](https://api.dartlang.org/stable/1.24.3/dart-async/Future/Future.html)).

Reply:
Aah, yeah, then it's the same. Huh. What a weird API choice.